app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -294,6 +271,17 @@ class NavigationOverlayFragment : Fragment() {
         ).map { channel -> channel.forwardRemoveEventsToRepo() }
     }
 
+    private fun observeToolbarState(): Disposable {
+        return navigationOverlayViewModel.leftmostActiveToolBarId
+                .subscribe { viewToFocus ->
+                    // Reset previous left most active toolbar button's nextFocusLeftID
+                    rootView?.findViewById<View>(navUrlInput.nextFocusUpId)?.nextFocusLeftId = -1
+                    rootView?.findViewById<View>(viewToFocus)?.nextFocusLeftId = viewToFocus
+
+                    navUrlInput.nextFocusUpId = viewToFocus
```

Comment:
Nit: the logic here is pretty simple, but there's a lot of noise that makes it a little hard to read.  I think a comment between 278 & 279 would help for clarity, something like `// Disable left clicks on viewToFocus `